tests/integration/test_postgresql_database_engine/test.py

```diff
@@ -65,6 +65,15 @@ def test_postgres_database_engine_with_postgres_ddl(started_cluster):
 
     create_postgres_table(cursor, "test_table")
     assert "test_table" in node1.query("SHOW TABLES FROM postgres_database")
+    create_postgres_table(
+        cursor,
+        "test_table",
+        suffix=""" SETTINGS
+        connection_pool_size = 50,
```

Comment:
let's also add 3 tests:
1. set `connection_pool_size = 0` in `SETTINGS` and check that create query throws exception "connection pool size cannot be zero"
2. set `connection_pool_size = 0` in named collections and NOT add any `SETTINGS` to query and check that create query throws exception "connection pool size cannot be zero"
3. set `connection_pool_size = 0` in named collections and DO add  `SETTINGS connection_pool_size = 10` and check that exception is NOT thrown, because we want SETTINGS to override settings from named collections.

Reply:
When I run in client:
```
CREATE TABLE default.postgresql_table
(
    `float_nullable` Nullable(Float32),
    `str` String,
    `int_id` Int32
)
ENGINE = PostgreSQL('localhost:5432', 'public', 'test', 'postges', '') SETTINGS connection_pool_size = 0, connection_pool_auto_close = true ;
```
I get back success. I'm trying to attach debugger to see if it stepped through the check in `StoragePostgreSQL.cpp:557` with either `sudo lldb -p pgrep clickhouse` or `sudo lldb clickhouse server` or gdb but it doesn't work for me.
Looking at `StoragePostgreSQL.cpp` I'm only checking `postgresql_connection_pool_size` through common settings. I'm still unsure how to properly check the PostgreSQL specific settings there, could you help me @kssenii ?

Reply:
> Also, what's the difference between the first 2 tests? I assume in 2nd case we have the settings dict with only KV for connection_pool_size, whereas in 1st case we have other settings as well?
In the 3rd case, we would set the same key twice with different values? That wouldn't be doable by passing a dict as argument to create_postgres_table right?


**test 1** (just a check that `SETTINGS` clause works):
```
assert "postgresql_connection_pool_size cannot be zero" in node1.query_and_get_error(
    "CREATE TABLE (...) ENGINE PostgreSQL = (...) SETTINGS connection_pool_size = 0"
)
```

**test 2** (since you added support for PostgreSQLSettings through named collections):
```
assert "postgresql_connection_pool_size cannot be zero" in node1.query_and_get_error(
    "CREATE TABLE (...) ENGINE PostgreSQL = (postgres_conf, connection_pool_size = 0)"
)
```

**test 3** (check `SETTINGS` clause can override settings from named collections):
```
# no exception
node1.query(
    "CREATE TABLE (...) ENGINE PostgreSQL = (postgres_conf, connection_pool_size = 0) SETTINGS connection_pool_size = 10"
)
# exception
assert "postgresql_connection_pool_size cannot be zero" in node1.query_and_get_error(
    "CREATE TABLE (...) ENGINE PostgreSQL = (postgres_conf, connection_pool_size = 10) SETTINGS connection_pool_size = 0"
)
```

Reply:
>  I'm trying to attach debugger to see if it stepped through the check in StoragePostgreSQL.cpp:557

why do you need to check this in debugger if in line `StoragePostgreSQL.cpp:557` we throw exception if check did not pass, so all you need is do a query `SETTINGS connection_pool_size = 0` in order to check what you want?

Reply:
I think I'm missing a check of `postgresql_settings` after calling `StoragePosgreSQL::getConfiguration(args.engine_args, args.getLocalContext(), postgresql_settings)`, for the check against `connection_pool_size`. That should give correct error.

I'm waiting for ninja to finish to test this out.

Reply:
So inside `PostgreSQLSettings::loadFromQuery(ASTStorage &)`, we should set:
```cpp
storage_def.settings->changed = !storage_def.settings->changes.empty();
```
before returning, right?

Reply:
no sorry, I double checked, my assumption in previous comment was incorrect, `loadFromQuery` sets only actually changed settings.